### DISCUSSION TITLE
StarTrail Videos not saving

### DISCUSSION THREAD

**Discussion by jshall77** [aaronwmorris/indi-allsky#957] [Q&A]
Hello, I am having another issue where the startrail videos say they are being generated and show the thumbnail preview, but cannot be watched, downloaded and is also not being copied over to my FTP site like all the other images and videos.  Has anyone else had this problem?

**Comment by aaronwmorris** [comment]
The default settings require a minimum of 250 startrail frames before a video is generated (this is configurable).  Being that we are close to a full moon, it is likely you may not have enough frames for one to be generated.  There is always a preview created, no matter how many frames are available.

The logs should tell you if there was a problem with it being generated or not enough frames were available.

**Comment by jshall77** [comment]
If this is the case, why is there still a link to the startrail video and
thumbnail? Shouldn’t it send a notification or show as that it’s not being
generated in the logs?

**Reply by aaronwmorris** [reply]
Sorry, I just made a bad assumption.  I would check the logs.  You can grep for the Video process messages to narrow things down.

```
grep Video /var/log/indi-allsky/indi-allsky.log | less
```

You can also delete the startrails (and keogram) and try to regenerate them under the timelapse generator view.  You can easily see when the generation failed because there will be an [!] exclamation mark in the drop down.

Strangely enough, I have had about a week where one of my cameras was having trouble generating videos about half the time.  What version of ffmpeg do you have installed?

```
dpkg -l|grep ffmpeg
```

**Reply by jshall77** [reply]
Thanks for the quick reply Aaron!  Here is the output from the above:
ii  ffmpeg                               8:4.3.6-0+deb11u1+rpt5           arm64

It appears that the startrail videos are being generated as I can see them in /var/html/allsky/images, however, when I try to view them in the browser, I just get the video window and it doesn't allow me to play it.  I copied a couple over to my PC using WinSCP, but even VLC and other video players cannot play them.  

I also realized that somehow even though I only have about a week's worth of photos and videos, my 256GB SD card is full(separate issue).

**Reply by aaronwmorris** [reply]
Its possible you may just be running out of space.  I would estimate that it takes about 2-4GB of free space to store the temporary images needed to generate the startrails.

I would try to manually generate the files again and see if more verbose error messages are shown in the logs

**Reply by jshall77** [reply]
So the space issue was actually caused by the timelapse videos failing and leaving a few folders full of duplicate images, which eventually filled up the drive.  I cleaned that up, but the startrail timelapse are still failing.  I do have moon mode turned off, but they are still not working.  I rebooted this morning and am re-running it right now, but it will most likely fail again.  Let me know what you would like to see from the logs.  Thanks again!

**Comment by jshall77** [comment]
![image](https://github.com/aaronwmorris/indi-allsky/assets/79818454/1daf8968-b9d5-4007-a220-814583c94e08)

**Reply by aaronwmorris** [reply]
I may be underestimating the amount of space needed to generate the startrails timelapse.  It might require more than 4GB of free disk space with a 4K camera.

The other possibility is you may be running out of memory generating videos.  Generating videos is very memory intensive and scales linearly with resolution, ie more resolution needs more memory.  4K videos can run up against the memory of a 4GB Raspi 4.  You may want to set the video scaling to 75%.

**Reply by aaronwmorris** [reply]
Addendum:  If you are running out of memory, you will not receive any log messages in indi-allsky indicating that.  The out of memory condition will trigger the kernel to kill the process using the most memory.  indi-allsky does not know why the process stopped, just that it no longer exists.  You will have to check the system logs to see if you see any OOM related messages.
